File: spring/src/main/java/com/givoo/entity/Inquiry.java
package com.givoo.entity;

import jakarta.persistence.*;
import lombok.*;

import java.text.SimpleDateFormat;
import java.util.Date;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@ToString
@Table(name = "inquiry")
public class Inquiry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "inquiry_id")
    private Long inquiryId;

    @Column(name = "title", nullable = false)
    private String title;

    @Lob
    @Column(name = "content", nullable = false)
    private String content;

    @Column(name = "category")
    private String category;

    @Column
    private Long userId;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(name = "created_at", updatable = false)
    private String createdAt;
    @PrePersist
    protected void onCreate() {
        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy.MM.dd");
        createdAt = dateFormat.format(new Date());
    }

    @Column(name = "status")
    private String status;

    @Column(name ="answer")
    private String answer;
    @Column(name="answer_created")
    private String answerCreated;
    public Inquiry(Long userId, String title, String content) {
        this.userId =userId;
        this.title =title;
        this.content=content;
        this.status="0";
    }

}

File: flutter/lib/pages/mypage/model/Inquiry.dart
import 'dart:convert';

List<Inquiry> inquiryFromJson(String str) => List<Inquiry>.from(json.decode(str).map((x) => Inquiry.fromJson(x)));

String inquiryToJson(List<Inquiry> data) => json.encode(List<dynamic>.from(data.map((x) => x.toJson())));

class Inquiry {
  int inquiryId;
  String title;
  String content;
  dynamic category;
  int userId;
  String createdAt;
  String status;
  dynamic answer;
  dynamic answerCreated;
  Inquiry({
    required this.inquiryId,
    required this.title,
    required this.content,
    required this.category,
    required this.userId,
    required this.createdAt,
    required this.status,
    required this.answer,
    required this.answerCreated,
  });

  factory Inquiry.fromJson(Map<String, dynamic> json) => Inquiry(
    inquiryId: json["inquiryId"],
    title: json["title"],
    content: json["content"],
    category: json["category"],
    userId: json["userId"],
    createdAt: json["createdAt"],
    status: json["status"],
    answer: json["answer"],
    answerCreated: json["answerCreated"],
  );

  Map<String, dynamic> toJson() => {
    "inquiryId": inquiryId,
    "title": title,
    "content": content,
    "category": category,
    "userId": userId,
    "createdAt": createdAt,
    "status": status,
    "answer": answer,
    "answerCreated":answerCreated,
  };
}